crates/matrix-sdk-crypto/src/olm/session.rs

```diff
@@ -126,11 +126,9 @@ impl Session {
     /// * `plaintext` - The plaintext that should be encrypted.
     pub(crate) async fn encrypt_helper(&mut self, plaintext: &str) -> OlmMessage {
         let mut session = self.inner.lock().await;
-
-        Span::current().record("session", debug(&session));
         let message = session.encrypt(plaintext);
-
         self.last_use_time = SecondsSinceUnixEpoch::now();
+        debug!(?session, "Successfully encrypted an event");
```

Comment:
I'm deliberately moving this field from the span, to the individual event. Firstly, it's quite a lot of data and, if there *were* more than one event in this span, we wouldn't want to log it on every event. More importantly, it's important that we capture the state of the session *after* the encryption happens (we want to know if it's stepped the DH ratchet, etc.)  Given that the state of the `session` changes over the lifetime of the span, it feels inappropriate to attach it to the span.